app/models/audio_version.rb

```diff
@@ -20,4 +25,33 @@ def length(reload=false)
   def self.policy_class
     StoryAttributePolicy
   end
+
+  def compliant_with_template?
+    file_errors.nil?
+  end
+
+  private
+
+  def validate_on_template
+    return unless audio_version_template
+
+    noncompliant_file = audio_files.find do |af|
+      !af.compliant_with_template? || af.status == INVALID
+    end
+    if noncompliant_file
+      self.status = INVALID
+      self.file_errors = noncompliant_file.audio_errors ||
```

Comment:
Alternately, could comma-join every file error here, instead of finding the first.